src/lib/lib.rs

```diff
@@ -10,27 +10,34 @@ pub mod syntax;
 #[cfg(feature = "wasm-bindgen")]
 mod wasm;
 
+#[cfg(feature = "wasm-bindgen")]
+pub use crate::wasm::*;
 use crate::{
     builtins::value::ResultValue,
     exec::{Executor, Interpreter},
     realm::Realm,
     syntax::{ast::expr::Expr, lexer::Lexer, parser::Parser},
 };
-#[cfg(feature = "wasm-bindgen")]
-pub use wasm::*;
 
-fn parser_expr(src: &str) -> Expr {
+fn parser_expr(src: &str) -> Result<Expr, String> {
     let mut lexer = Lexer::new(src);
-    lexer.lex().expect("lexing failed");
+    lexer.lex().map_err(|e| e.to_string())?;
     let tokens = lexer.tokens;
-    Parser::new(tokens).parse_all().expect("parsing failed")
+    Parser::new(tokens)
+        .parse_all()
+        .map_err(|_e| "parsing failed".to_string())
```

Comment:
Then this can say ParsingError: {e.to_string()}

Reply:
I had to give `ParseError` a basic `Display` impl, but I figured that it was needed eventually and should be able to be customized better later.